github-oidc/README.md

```diff
@@ -1,63 +1,85 @@
 
-`github-oidc/main.tf`
+## IAM Role
 
-```hcl
-  module "github-actions-runner-aws" {
-    source = "github.com/CMSgov/github-actions-runner-aws//github-oidc" # double-slash denotes a sub-directory
+This is the role that the Actinos workflow assumes via the OIDC request. The permissions defined for this role dictate the scope of the short-term credentials that are generated and passed back to GitHub by AWS. The ARN of this role is output by each of the IaC methods listed, and is passed in to the [GitHub Action that configures AWS credentials for workflows](https://github.com/aws-actions/configure-aws-credentials#assuming-a-role). See the example below under "Using the OIDC Provider in a Workflow."
 
-    subject_claim_filters                         = ["repo:{your GitHub org}/{your GitHub repo}:*"]
-    # audience_list                               = [] # optional, defaults to ["sts.amazonaws.com"]
-    # thumbprint_list                             = [] # optional, defaults to ["6938fd4d98bab03faadb97b34396831e3780aea1"]
-    # github_actions_permissions_policy_json_path = "" # optional, defaults to "github_actions_permission_policy.json"
-    # add_read_only_access                        = bool # optional, defaults to false
-  }
-```
+### IAM Trust Policy
 
-`github-oidc/github_actions_permission_policy.json`
+This policy establishes the trust relationship between GitHub and AWS based on values in the JSON web token (JWT) submitted by the GitHub OIDC provider. Here's an example of the JWT sent by GitHub to AWS:
 
 ```json
 {
-  "Version": "2012-10-17",
-  "Statement": [
-    {
-      "Action": ["securityhub:BatchImportFindings"],
-      "Effect": "Allow",
-      "Resource": "*"
-    },
-    {
-      "Sid": "UpdateService",
-      "Effect": "Allow",
-      "Action": ["ecs:UpdateService"],
-      "Resource": [
-        "arn:aws:ecs:{your region}:{your account number}:service/{your self-hosted runner cluster name}/{your github runner service name}"
-      ]
-    }
-  ]
+  "typ": "JWT",
+  "alg": "RS256",
+  "x5t": "example-thumbprint",
+  "kid": "example-key-id"
+}
+{
+  "jti": "2e624367-5ab9-4133-c537-861d1b19a85b",
+  "sub": "repo:{organization}/{repo}:ref:refs/heads/{branch}",
+  "aud": "sts.amazonaws.com",
+  "ref": "refs/heads/{branch}",
+  "sha": "ec7d38be2362bfaaf8878a1cebb4b0f695eab764",
+  "repository": "{organization}/{repo}",
+  "repository_owner": "{organization}",
+  "repository_owner_id": "3209407",
+  "run_id": "3339113236",
+  "run_number": "117",
+  "run_attempt": "3",
+  "repository_visibility": "public",
+  "repository_id": "376113068",
+  "actor_id": "25254258",
+  "actor": "{github username}",
+  "workflow": "Run security scans",
+  "head_ref": "",
+  "base_ref": "",
+  "event_name": "push",
+  "ref_type": "branch",
+  "job_workflow_ref": "{organization}/{repo}/.github/workflows/security.yml@refs/heads/bharvey-security-skip-build",
```

Comment:
I don't think it matters all that much but should this reference the main branch?

Reply:
Good call, I generalized a couple of other specific things as well.